frame/3/bli_l3_thrinfo.c

```diff
@@ -77,7 +77,18 @@ void bli_l3_thrinfo_grow
 	thrinfo_t* thread_cur = bli_thrinfo_split( n_way, thread_par );
 	bli_thrinfo_set_sub_node( thread_cur, thread_par );
 
-	if ( sub_prenode != NULL )
+    if ( bszid == BLIS_NO_PART )
```

Comment:
I wonder if there is a replacement to the condition `if ( bszid == BLIS_NO_PART )` (there are tons of it in BLIS) to something clearer, like:

```lang=c
const bool doing_packing = ...; // perhaps giving a new name to BLIS_NO_PART to mean "packing" ?
if ( doing_packing )
{
    ...
}
```

PS: you seem to have indent issues in some places, or it was the Github viewer ?

Reply:
We have lots of indent issues because I write spaces and then Field changes them to tabs. It's a process.

Reply:
But this is really a hack for now to fix performance and potential correctness issues. This will be replaced by something more robust eventually.

Reply:
Although do note that the only control tree nodes with BLIS_NO_PART at the moment are packing nodes.

Reply:
> Although do note that the only control tree nodes with BLIS_NO_PART at the moment are packing nodes.

Yes, that's why I suggested to rename it to something more understandable. I recall myself the first time reading the code to find out what does `BLIS_NO_PART` mean.

Reply:
I was trying to future-proof the `enum` when naming it `BLIS_NO_PART` because I envisioned maybe someday there would be an operation that doesn't partition but doesn't pack either. I agree it's not super clear and I'm 100% open to it being improved.